scrapy/http/request/form.py

```diff
@@ -107,8 +108,13 @@ def _get_inputs(form, formdata, dont_click, clickdata, response):
 
     inputs = form.xpath('descendant::textarea'
                         '|descendant::select'
-                        '|descendant::input[@type!="submit" and @type!="image" and @type!="reset"'
-                        'and ((@type!="checkbox" and @type!="radio") or @checked)]')
+                        '|descendant::input[not(@type) or @type['
+                        ' translate(., "SUBMIT", "submit") != "submit"'
```

Comment:
`re:test(@type, "^(submit|image|reset)$", "i")` only for the first 3
and separately the checkbox and the radio
Alternatively it's a regexp per value.
I didn't test it but I think there should be a performance penalty
invoking python's re module instead of leaving it to libxml.